numba/np/arraymath.py

```diff
@@ -4192,41 +4196,69 @@ def op_nop(x):
             a_op = op_conj
             b_op = op_nop
 
-    def impl(a, v):
+    _NP_PRED = numpy_version > (1, 17)
+
+    def impl(a, v, mode="valid"):
         la = len(a)
         lv = len(v)
-        if la == 0:
-            raise ValueError("'a' cannot be empty")
-        if lv == 0:
-            raise ValueError("'v' cannot be empty")
+
+        if mode == "full":
+            corr_mode = Mode.FULL
```

Comment:
Since we're now introducing the `mode` as an argument I think the `Mode` Enum framework can be removed entirely.

We can simply pass the `mode` in the `_np_correlate_core` and compare it over there as a string?